pytorch_tabnet/tab_network.py

```diff
@@ -405,3 +376,71 @@ def forward(self, x):
         x = self.bn(x)
         out = torch.mul(x[:, :self.output_dim], torch.sigmoid(x[:, self.output_dim:]))
         return out
+
+
+class EmbeddingGenerator(torch.nn.Module):
+    """
+        Classical embeddings generator
+    """
+
+    def __init__(self, input_dim, cat_dims, cat_idxs, cat_emb_dim):
+        """ This is an embedding module for an entier set of features
+
+        Parameters
+        ----------
+        input_dim : int
+            Number of features coming as input (number of columns)
+        cat_dims : list of int
+            Number of modalities for each categorial features
+            If the list is empty, no embeddings will be done
+        cat_idxs : list of int
+            Positional index for each categorical features in inputs
+        cat_emb_dim : int or list of int
+            Embedding dimension for each categorical features
+            If int, the same embdeding dimension will be used for all categorical features
+        """
+        super(EmbeddingGenerator, self).__init__()
+        if cat_dims == []:
+            self.skip_embedding = True
+            self.post_embed_dim = input_dim
+        else:
+            self.skip_embedding = False
+            if isinstance(cat_emb_dim, int):
+                self.cat_emb_dims = [cat_emb_dim]*len(cat_idxs)
+                self.post_embed_dim = int(input_dim + (cat_emb_dim - 1)*len(cat_idxs))
+            else:
+                # check that all embeddings are provided
+                assert(len(cat_emb_dim) == len(cat_dims))
```

Comment:
It would be simpler to do
```
            if isinstance(cat_emb_dim, int):
                self.cat_emb_dims = [cat_emb_dim]*len(cat_idxs)
           else:
                 self.cat_emb_dims = cat_emb_dim
```
And then, do the safety check on length, build post_embed_dim, ...